index.js

```diff
@@ -197,26 +214,90 @@ function uploadMergeFiles(viewId, mergeData, templatesToRecords){
         return Promise.all(uploadPromises);
     })
     .then((uploadResponses) =>{
-        console.log("done uploading everything");
+        log.log("done uploading everything");
+
         if(globalCallback){
             globalCallback(null, "Merge completed successfully");
         }
     })
     .catch(function(err) {
-        handleError(err);
+      checkFieldNames("merged", config.merged_doc_table.view_id);
+      handleError(err);
     });
 }
 
 /**
+ * Function to determine what field is causing the upload error
+ * @param {String} table
+ * @param {Number} viewId
+ */
+function checkFieldNames(table, viewId) {
+  promises = [];
+  promises.push(api.getView(viewId));
+  Promise.all(promises)
+  .then((view) => {
+    if ( promises.length > 0 &&
+        table == "merged" &&
+        !findDocFieldName(view[0].structure, config.merged_doc_table.merged_doc_details_field_name) &&
```

Comment:
You don't need to be exclusive about your errors. Right now if the user had two fields incorrectly set they'd never get an error message because it would match none of these ifs. Also if any field is set incorrectly then it should fail, regardless of if anything else is or is not incorrect.

I'd do this
```javascript
var structure = view[0].structure; //makes your code a lot cleaner
if(!findDocFieldName(structure, config.merged_doc_table.merged_doc_details_field_name)){ 
     log.error(`Unable to find details field from ${table} table, please ensure EXACT match`);
     return;
}
```